pyquil/api/_compiler.py

```diff
@@ -391,230 +230,44 @@ def get_calibration_program(self) -> Program:
-
-    def connect(self) -> None:
-        try:
-            version_dict = self.get_version_info()
-            check_quilc_version(version_dict)
-        except TimeoutError:
-            raise QuilcNotRunning(
-                f"Request to quilc at {self.client.endpoint} timed out. "
-                "This could mean that quilc is not running, is not reachable, or is "
-                "responding slowly."
-            )
-
-    def get_version_info(self) -> Dict[str, Any]:
-        return cast(Dict[str, Any], self.client.call("get_version_info"))
+        super().__init__(device=device, client=client, timeout=timeout)
 
     @_record_call
-    def quil_to_native_quil(self, program: Program, *, protoquil: Optional[bool] = None) -> Program:
-        self.connect()
-        request = NativeQuilRequest(quil=program.out(), target_device=self.target_device)
-        response = self.client.call("quil_to_native_quil", request, protoquil=protoquil).asdict()
-        nq_program = parse_program(response["quil"])
-        nq_program.native_quil_metadata = response["metadata"]
-        nq_program.num_shots = program.num_shots
-        return nq_program
-
-    @_record_call
-    def native_quil_to_executable(self, nq_program: Program) -> PyQuilExecutableResponse:
+    def native_quil_to_executable(self, nq_program: Program) -> QuantumExecutable:
         return PyQuilExecutableResponse(
             program=nq_program.out(),
             attributes=_extract_attribute_dictionary_from_program(nq_program),
         )
-
-    @_record_call
-    def reset(self) -> None:
-        """
-        Reset the state of the QVMCompiler quilc connection.
-        """
-        timeout = self.client.timeout
-        self.client.close()  # type: ignore
-        self.client = Client(self.endpoint, timeout=timeout)
-
-    @property
-    def quilc_client(self) -> Client:
-        """Return the `Client` for the compiler (i.e. quilc, not translation service)."""
-        return self.client
-
-    def set_timeout(self, timeout: float) -> None:
-        """
-        Set timeout for each individual stage of compilation.
-
-        :param timeout: Timeout value for each compilation stage, in seconds. If the stage does not
-            complete within this threshold, an exception is raised.
-        """
-        if timeout < 0:
-            raise ValueError(f"Cannot set timeout to negative value {timeout}")
-
-        self.timeout = timeout
-        self.quilc_client.rpc_timeout = timeout
-
-    @_record_call
-    def get_calibration_program(self) -> Program:
-        """
-        See ``QPUCompiler.get_calibration_program()``.
-
-        Note: this currently provides an empty Program because the QVM does not support Quil-T.
-        """
-        return Program()
-
-    @_record_call
-    def refresh_calibration_program(self) -> None:
-        pass
-
-    @property
-    def calibration_program(self) -> Program:
-        return Program()
-
-
-@dataclass
-class HTTPCompilerClient:
```

Comment:
This class was removed altogether because it was used for compiling to binary executables, which is now a normal HTTP request and therefore handled by calling `api.Client.qcs_request()` (see `QPUCompiler`)